py3status/modules/nvidia_temp.py

```diff
@@ -43,33 +59,21 @@ def nvidia_temp(self):
         if temps != []:
             data = []
             for temp in temps:
-                fmt_str = "{temp}{format_units}".format(
-                    temp=temp,
-                    format_units=self.format_units
-                )
-                data.append(fmt_str)
-
-            output = "{format_prefix}{data}".format(
-                format_prefix=self.format_prefix,
-                data=self.temp_separator.join(data)
-            )
+                data.append(self.py3.safe_format(self.gpu_temp, {'temp': temp}))
 
+            output = "{data}".format(data=self.icon_separator.join(data))
```

Comment:
here we want to build a composite to pass as `{gpu_temps}` which is basically your data but with `{'full_text': self.icon_separator}` between each item.

probably easiest to add during your loop and then remove the last item.

maybe in future the separator should be a format so we can colour it? for now I'd just keep it plain text.